tests/e2e/e2e.test.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @file Contains end-to-end tests for the vercel preview instance.
+ */
+import { describe } from "@jest/globals";
+import axios from "axios";
+import dotenv from "dotenv";
+dotenv.config();
+
+describe("Fetch Cards", () => {
+  let VERCEL_PREVIEW_URL;
+
+  beforeAll(() => {
+    VERCEL_PREVIEW_URL = process.env.VERCEL_PREVIEW_URL;
+  });
+
+  test("retrieve stats card", async () => {
+    expect(VERCEL_PREVIEW_URL).toBeDefined();
+    await expect(
+      axios.get(`${VERCEL_PREVIEW_URL}/api?username=willianrod`),
```

Comment:
It could still return but return error. We should test if it's actually returning svg with proper stats.

Reply:
@anuraghazra Good point. I updated the code a bit so that it now checks the snapshot. While coding, I realised this would break if the stats of the used user accounts (i.e.`anuraghazra` and `willianrod`) are updated and if the code creating the cards is updated. 

I could create a dummy GitHub account in which the stats stay static; however, in that case, the test will still fail when the card generation code is updated. The best solution I could think of now is to generate the cards based on local data and create a static user account for which this data does not change. A bit similar to the regular tests, but then we compare the server SVG with the locally generated SVG.

https://github.com/anuraghazra/github-readme-stats/blob/fe1ca87e33cfbfd75b058fe52bfe7c3e15e2c8c4/tests/renderStatsCard.test.js#L25

Maybe you have a better idea? 🤔

Reply:
@anuraghazra I implement the solution above, which seems to work (see https://github.com/anuraghazra/github-readme-stats/actions/runs/3142651570/jobs/5106496325).